[Ong Jun Jie] iP

junjunjieOng opened the pull request.

Commit:
Add Gradle support

junjunjieOng committed:
duke chatbot

junjunjieOng committed:
Add add feature and list

junjunjieOng committed:
Add mark and unmark features

junjunjieOng committed:
Add three types of tasks

junjunjieOng committed:
Update test cases and refine code

junjunjieOng committed:
Add handle errors feature and update test cases

junjunjieOng committed:
Add handle errors feature and update test cases

junjunjieOng committed:
Add delete feature

junjunjieOng committed:
Add use of enums

junjunjieOng committed:
Update Deadline.java, Task.java, Event.java, ToDo.java, Duke.java

junjunjieOng committed:
Add date and time function for Deadline.java, Duke.java, Event.java, Task.java

junjunjieOng committed:
Merge branch 'branch-Level-8'

junjunjieOng committed:
Refactor code based on OOP

junjunjieOng committed:
Divide the classes into different packages

junjunjieOng committed:
Add JUnit test

junjunjieOng committed:
Add JavaDoc comments

junjunjieOng committed:
Tweak the code to comply with a coding standard

junjunjieOng committed:
Add find features

junjunjieOng committed:
Merge tag 'Level-9'

junjunjieOng committed:
Resolve merge conflicts for A-JavaDoc, A-CodingStandard, Level-9

junjunjieOng committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

junjunjieOng committed:
Add Gradle support

junjunjieOng committed:
Add checkstyle and corrected coding style error

junjunjieOng committed:
Add GUI for dukebot

weijuey commented on src/main/java/Duke/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,58 @@
+        try {
+            int currTaskNum = Integer.parseInt(taskNum);
+            if (tasks.getTaskList().size() >= currTaskNum && currTaskNum > 0) {
+                Task currTask = tasks.getTaskList().get(currTaskNum - 1);
+                tasks.deleteTask(currTaskNum - 1);
```
I noticed that the markTask method returned the task, would it better to do the same with deleteTask here?

weijuey commented on src/main/java/Duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+     * @param input input of the keyword by user.
+     * @return the list of result that match the input.
+     */
+    public ArrayList<Task> findTask(String input) {
```
I think this method can be named better, other methods like markTask and deleteTask are operations on 1 task and returns 1 task, if any, but findTask, although named the same way, returns a collection of tasks. Pluralising Task should be good enough.

weijuey commented on src/main/java/Duke/ui/Ui.java:
```diff
@@ -0,0 +1,130 @@
+     *
+     * @param resultLists result lists from the search keyword.
+     */
+    public String displaySearchResult(ArrayList<Task> resultLists) {
```
I think `resultList` or `results` are better names for this parameter.

weijuey reviewed (commented):
Looks great, I could understand your code very easily, naming and coding style requirements were abided to almost perfectly. Just a few nits.

ngjunkang commented on src/main/java/Duke/storage/Storage.java:
```diff
@@ -0,0 +1,95 @@
+                String[] currTaskLine = currLine.split("\\|");
+                switch (currTaskLine[0]) {
+                case "T":
+                    ToDo todoTask = new ToDo(currTaskLine[2]);
```
Is ToDo one-word or two-word? The class `ToDo` seems to be two-word, however the variable name `todoTask` seems to be one-word :)

ngjunkang commented on src/main/java/Duke/tasklist/TaskList.java:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    /**
+     * Marks the task in the task list as complete.
+     *
+     * @param taskNum the task number in the list.
+     * @return the task that is mark.
```
Did you forget to edit your Javadoc method header?

ngjunkang commented on src/main/java/Duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    /**
+     * Strings representation of Deadline in the save file.
```
I feel that `Strings` is not widely used as a verb. A better way to describe this will be `Returns the string representation of ...`. I've noted the same issue in other places too.

ngjunkang commented on src/main/java/Duke/tasks/Task.java:
```diff
@@ -0,0 +1,108 @@
+        isDone = false;
+    }
+
+    public abstract String saveToFileString();
```
`saveToFileString` sounds like you are saving something inside the method, as the action verb is `save`. I would name this as `encodeTaskToString`.

ngjunkang reviewed (commented):
Good job! LGTM, just some nits to fix. 👍

joszx commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,51 @@
+        }
+    }
+
+    /**
+     * Returns the response base on the input.
+     */
```
I believe there should be @param and @return in this javadocs?

joszx commented on src/main/java/Duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,44 @@
+     * @param tasks List of the tasks.
+     * @param ui UI that deals with interactions with the user.
+     * @param storage storage handles the saving and writing to file.
+     * @return message stating task adeed.
```
```suggestion
     * @return message stating task added.
```
Slight mispell?

joszx commented on src/main/java/Duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,44 @@
+     *
+     * @return false that user not exiting.
+     */
+    public boolean isExit() {
```
Perhaps isExiting would be a more suitable method name?

joszx commented on src/main/java/Duke/storage/Storage.java:
```diff
@@ -0,0 +1,95 @@
+     * @param taskList the list of tasks.
+     * @throws DukeException if you cannot write into the file.
+     */
+    public void save(ArrayList<Task> taskList) throws DukeException {
```
Same as in the load method.

joszx commented on src/main/java/Duke/storage/Storage.java:
```diff
@@ -0,0 +1,95 @@
+     * @return the the list of tasks.
+     * @throws DukeException If files cannot be read or created.
+     */
+    public ArrayList<Task> load() throws DukeException {
```
Maybe a more specific method name like loadTaskListFromStorage? Assuming the storage could be used to load other things in the future :)

joszx reviewed (commented):
LGTM. Agree with @weijuey. Coding standard was followed to near perfection. Had to really dig to find some nits. Cheers! 👍

junjunjieOng committed:
Add assertion to document important assumptions that should hold

# Methods usually assume the input taken are correct
# Assertion can catch those mistake so that the runtime can verify them
# Assertion provide more efficient detection and correction of bugs in the code

junjunjieOng committed:
Merge pull request #1 from junjunjieOng/branch-A-Assertions

Add assertion to document important assumptions that should hold

junjunjieOng committed:
Refactor and improve code quality

Some cases uses similar codes to do the same checks.
Unifying those duplicated codes into a method helps to improves the code quality.
Let's extract those duplicated code and separate them to a private method. Doing so will
help to understand the code better.

junjunjieOng committed:
Merge branch 'master' into branch-A-CodeQuality

junjunjieOng committed:
Merge pull request #2 from junjunjieOng/branch-A-CodeQuality

Refactor and improve code quality

junjunjieOng committed:
Fix function error

junjunjieOng committed:
Create gradle.yml

junjunjieOng committed:
Add the ability to detect duplicate task input

junjunjieOng committed:
Improve the GUI of duke

junjunjieOng committed:
Add a User Guide

junjunjieOng committed:
Rectify minor error

junjunjieOng committed:
Add UserGuide

junjunjieOng committed:
Set theme jekyll-theme-cayman
